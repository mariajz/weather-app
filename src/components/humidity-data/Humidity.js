import React from 'react';
import WeatherDetailsCard from '../../commons/weather-details-card';
import { getHumidityText } from '../../utils/helpers';

const Humidity = ({ humidityProps }) => {
    const { humidity } = humidityProps;
    const { level, humidityText } = getHumidityText(humidity);
    const humidityCardProps = {
        iconName: 'humidity',
        cardName: 'HUMIDITY',
        title: `${humidity}%`,
        subtitle: level,
        content: humidityText,
    };

    return <WeatherDetailsCard weatherDetailsCardProps={humidityCardProps} />;
};

export default Humidity;
