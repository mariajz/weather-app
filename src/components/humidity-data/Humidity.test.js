import { render } from '@testing-library/react-native';
import React from 'react';
import Humidity from './Humidity';

const defaultProps = {
    humidityProps: {
        humidity: 70,
    },
};

describe('Humidity', () => {
    it('should render Humidity', () => {
        const container = render(<Humidity {...defaultProps} />);

        expect(container).toMatchSnapshot();
    });
});
