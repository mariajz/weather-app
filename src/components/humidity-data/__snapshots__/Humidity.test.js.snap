// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Humidity should render Humidity 1`] = `
<View
  style={
    {
      "backgroundColor": "rgba(255,255,255, 0.2)",
      "borderBottomLeftRadius": 10,
      "borderBottomRightRadius": 10,
      "borderTopLeftRadius": 10,
      "borderTopRightRadius": 10,
      "height": 170,
      "marginTop": 10,
      "paddingBottom": 20,
      "paddingLeft": 10,
      "paddingRight": 10,
      "paddingTop": 10,
      "width": "49%",
    }
  }
>
  <View
    style={
      {
        "flexDirection": "row",
      }
    }
  >
    <Image
      size={15}
      source={
        {
          "testUri": "../../../assets/icons/humidity.png",
        }
      }
      style={
        {
          "height": 15,
          "width": 15,
        }
      }
      testID="icon"
      tintColor="white"
    />
    <Text
      marginLeft={5}
      size={14}
      style={
        {
          "color": "white",
          "fontSize": 14,
          "marginLeft": 5,
        }
      }
    >
      HUMIDITY
    </Text>
  </View>
  <View
    style={
      {
        "height": 120,
        "justifyContent": "space-between",
        "marginTop": 8,
      }
    }
  >
    <View>
      <Text
        size={28}
        style={
          {
            "color": "white",
            "fontSize": 28,
            "marginLeft": 0,
          }
        }
      >
        70%
      </Text>
      <Text
        size={20}
        style={
          {
            "color": "white",
            "fontSize": 20,
            "marginLeft": 0,
          }
        }
      >
        High
      </Text>
    </View>
    <View />
    <Text
      size={10}
      style={
        {
          "color": "white",
          "fontSize": 10,
          "marginLeft": 0,
        }
      }
    >
      Be sure to stay cool, hydrated, and seek shade if possible.
    </Text>
  </View>
</View>
`;
